projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -17,3 +23,34 @@ RADIUS::RADIUS()
 {
 }
 
+int RADIUS::Start()
+{
+    isRunning = true;
+    m_thread = std::jthread([this](){ Run(); });
+    return 0;
+}
+
+int RADIUS::Stop()
+{
+    std::lock_guard lock(m_mutex);
+    isRunning = false;
```

Comment:
`isRunning` must be set to `false` in the `Run` method, before exiting.

Also, see the `ping` plugin and look how `stop_token` and `m_thread.request_stop()` are used. In particular, notice that we join the thread in `Stop` after requesting stop (or detaching it in case we failed to stop the thread function).